featurebyte/migration/service/mixin.py

```diff
@@ -183,19 +143,14 @@ async def get_session(self, feature_store: FeatureStoreModel) -> BaseSession:
         -------
         BaseSession
         """
-        credential_provider = MongoBackedCredentialProvider(persistent=self.persistent)
-        user = User(id=feature_store.user_id)
-        session_validator_service = SessionValidatorService(
-            user, self.persistent, self.catalog_id, credential_provider, self
-        )
-        session_manager_service = SessionManagerService(
-            user=user,
-            persistent=self.persistent,
-            catalog_id=self.catalog_id,
-            mongo_backed_credential_provider=credential_provider,
-            session_validator_service=session_validator_service,
-        )
-        session = await session_manager_service.get_feature_store_session(
+        feature_store_user = User(id=feature_store.user_id)
+        # A hack to update the user ID in the session manager service and session validator service to use
+        # the feature store user ID instead of the arbitrary current user ID. This is required for these
+        # services as we need to be able to retrieve the correct credentials from the persistent layer.
+        # We can use an arbitrary user ID for other services as there's no dependency on user ID.
+        self.session_manager_service.user = feature_store_user
+        self.session_manager_service.session_validator_service.user = feature_store_user
+        session = await self.session_manager_service.get_feature_store_session(
```

Comment:
@jevonyeoh Could we add a parameter in `self.session_manager_service.get_feature_store_session` to retrieve the credential based on provided `user_id`? If we override hte `user` here, not sure whether it will cause other issue if we run this on multiple threads.

@YS-L Once your PR (https://github.com/featurebyte/featurebyte/pull/1476) get merged, my understanding is that we don't need to migrate feature store metadata, right?

Reply:
good idea - updated!

Reply:
Maybe we could remove the comment above.

Reply:
While this solution works for now, I'm wondering if later similar issues will arise for other services as well. Adding user override all over the places seems not ideal. When that need arises, I think we have to revisit the way the services are instantiated within the migration services.